packages/devcmd/src/process/ProcessExecutor.ts

```diff
@@ -183,16 +228,25 @@ function childProcessCompletion(childProcess: ChildProcess): Promise<number | nu
   });
 }
 
-type Result = { ok: true } | { err: unknown };
+type Ok = { ok: true };
+type Err = { err: unknown };
+
+type Result = Ok | Err;
 
-function isErr(r: Result): r is { err: unknown } {
+function isErr(r: Result): r is Err {
   return !isOk(r);
 }
 
-function isOk(r: Result): r is { ok: true } {
+function isOk(r: Result): r is Ok {
   return "ok" in r && r.ok;
 }
 
+/**
+ * Executes an awaitable function and wraps the result of the Promise
```

Comment:
I think we should refer to the unwrap function and their combined uses here, otherwise this function isn't very meaningful.

maybe something like "... and wraps the result for later use with {@link unwrap}"

and please also add a similar docstring to the unwrap function :)